Javascript/100-Problems/script.js

```diff
@@ -850,3 +850,472 @@ const studentWithLowestPercentageInSubject = (subjectName) => {
 };
 
 console.log(studentWithLowestPercentageInSubject("Maths"));
+
+//function to find and print the subject(s) with the highest/lowest percentage of marks for a specific student
+
+const printSubjectPercentageOfStudent = (
+  studentName,
+  option,
+  obtainedPercentage
+) => {
+  const students = classObj.students;
+  const totalmark = 50;
+  let subjectName = "";
+  let subjectArray = [];
+  let subjectMark = 0;
+
+  students.find((student) => {
+    const subjects = student.marks;
+
+    if (student.name === studentName) {
+      subjectArray = subjects.map((subject) => {
+        subjectName = subject.subject;
+        subjectMark = subject.mark;
+
+        switch (option) {
+          case "highest":
+            if (obtainedPercentage <= subject.mark) {
+              obtainedPercentage = subject.mark;
+            }
+            break;
+
+          case "lowest":
+            if (obtainedPercentage >= subject.mark) {
+              obtainedPercentage = subject.mark;
+            }
+            break;
+
+          default:
+            return "Entered functionality not available! Please Enter the correct one";
+        }
+
+        const percentages = percentage(subjectMark, totalmark);
+        return { studentName, subjectName, percentages };
+      });
+    }
+  });
+
+  return subjectArray.filter(
+    (subject) =>
+      subject.percentages === percentage(obtainedPercentage, totalmark)
+  );
+};
+
+//Q41: Write a function to find and print the subject(s) with the highest percentage of marks for a specific student.
+
+const printSubjectWithHighestPercentageOfStudent = (studentName) => {
+  const option = "highest";
```

Comment:
this option is using many places. So that means we can make it a constant and use everywhere.